### PR TITLE
Add telegraf and go version to the internal input plugin

### LINKED ISSUE #5958: telegraf should be able to report its version
## Feature Request

As an operator, I should be able to check InfluxDB (or the backend datastore) to query/understand how many of each version of telegraf agents are reporting in.

### Use case: 
If I have 4,000 telegraf agents on 1.X and I'm performing an upgrade to 1.Y, I would like to ask/use InfluxDB to understand if all 4,000 agents are running 1.Y.  If they are not all 1.Y, how many and which are still 1.X.

### PR DESCRIPTION
**Pull request by adrianlzt**
Collecting telegraf version is useful to find outdated agents and match
behavioural changes with version changes.

closes #5958

### Required for all PRs:

- [x] Signed [CLA](https://influxdata.com/community/cla/).
- [x] Associated README.md updated.
- [ ] Has appropriate unit tests.

### CODE REVIEW

**Comment by alvarolmedo** [comment]
+1

File: plugins/inputs/internal/internal.go [hunk 1]
```diff
@@ -2,25 +2,31 @@ package internal
...
 
 	"github.com/influxdata/telegraf"
+	inter "github.com/influxdata/telegraf/internal"
 	"github.com/influxdata/telegraf/plugins/inputs"
 	"github.com/influxdata/telegraf/selfstat"
 )
 
 type Self struct {
 	CollectMemstats bool
+	SendVersion     bool
```
Comment (glinton) [inline plugins/inputs/internal/internal.go]: Deprecate `CollectMemstats` and use an include/exclude filter https://github.com/influxdata/telegraf/blob/4e59d51844dc2da4f2771eac4a9e33a986f8751b/plugins/inputs/docker_log/docker_log.go#L81-L82

https://github.com/influxdata/telegraf/blob/4e59d51844dc2da4f2771eac4a9e33a986f8751b/plugins/inputs/docker_log/docker_log.go#L398

**Comment by glinton** [comment]
Resolves #5958 
Resolves #5900

File: plugins/inputs/internal/README.md [hunk 2]
```diff
@@ -12,10 +12,18 @@ plugin.
 [[inputs.internal]]
   ## If true, collect telegraf memory stats.
   # collect_memstats = true
```
[the remark this answers is not in the document]
Comment (danielnelson) [reply plugins/inputs/internal/README.md]: Let's just skip the new option altogether, always send the new values.

File: plugins/inputs/internal/README.md [hunk 3]
```diff
@@ -12,10 +12,18 @@ plugin.
...
+  ## If true, send telegraf version
+  # send_version = true
 ```
 
 ### Measurements & Fields:
 
+Telegraf version and Go version used to compile it.
+
+- internal_version
```
[the remark this answers is not in the document]
Comment (danielnelson) [reply plugins/inputs/internal/README.md]: I have a different take on this, I'll comment on the other issue but it doesn't need to be part of this PR.  Here, let's add the Telegraf version as a tag on all existing metrics.  With the Go version, I'm tempted to say we should just leave it out, but I suppose it wouldn't be a problem to add it as a `go_version` tag on `internal_agent`.

```
internal_agent,host=tyrion,version=1.12.0,go_version=1.12.5 foo=42
internal_write,output=file,host=tyrion,version=1.12.0 foo=42
```

**Comment by glinton** [comment]
Just get the tests passing and this will be good to go

**Comment by raider111111** [comment]
Hi,

I updated to the latest version (1.12.0) but I'm not seeing the version tag. Is this only on specific counters?

The counters associated to the config:

```    [[inputs.win_perf_counters.object]]
     # The Telegraf Process Itself
     ObjectName = "Process"
     Instances = ["telegraf"]
     Counters = [
       "% Processor Time", "Working Set"
     ]
     Measurement = "win_process"
```


Suggestions?

**Comment by danielnelson** [comment]
The new tag is only added to the measurements from the `internal` input plugin.

**Pull request merged**